goalie/adjoint.py

```diff
@@ -512,11 +527,17 @@ def wrapped_solver(subinterval, initial_condition_map, **kwargs):
             # Solve adjoint problem
             tape = pyadjoint.get_working_tape()
             with PETSc.Log.Event("goalie.AdjointMeshSeq.solve_adjoint.evaluate_adj"):
-                m = pyadjoint.enlisting.Enlist(self._controls)
+                controls = pyadjoint.enlisting.Enlist(self._controls)
                 with pyadjoint.stop_annotating():
-                    with tape.marked_nodes(m):
+                    with tape.marked_nodes(controls):
                         tape.evaluate_adj(markings=True)
 
+                # Compute the gradient on the first subinterval
+                if i == 0 and compute_gradient:
+                    self._gradient = controls.delist(
+                        [control.get_derivative() for control in controls]
+                    )
+
```

Comment:
Note: the logic here is based on the logic in the Pyadjoint driver: https://github.com/dolfin-adjoint/pyadjoint/blob/dbf923e656b37333e45267da2e0264a84f160be5/pyadjoint/drivers.py#L5-L33.